core/state_processor.go

```diff
@@ -90,18 +70,18 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 		misc.ApplyDAOHardFork(statedb)
 	}
 
-	if p.random != nil && p.random.Running() {
-		err := p.random.RevealAndCommit(block.Randomness().Revealed, block.Randomness().Committed, header.Coinbase, header, statedb)
+	if random.IsRunning() {
+		err := random.RevealAndCommit(block.Randomness().Revealed, block.Randomness().Committed, header.Coinbase, header, statedb)
 		if err != nil {
 			return nil, nil, 0, err
 		}
 	}
 	// Iterate over and process the individual transactions
 	for i, tx := range block.Transactions() {
 		statedb.Prepare(tx.Hash(), block.Hash(), i)
-		infraFraction, _ := p.gpm.GetInfrastructureFraction(statedb, header)
-		gasPriceMinimum, _ := p.gpm.GetGasPriceMinimum(tx.GasCurrency(), statedb, header)
-		receipt, _, err := ApplyTransaction(p.config, p.bc, nil, gp, statedb, header, tx, usedGas, cfg, p.gcWl, p.regAdd, gasPriceMinimum, infraFraction)
+		infraFraction, _ := gpm.GetInfrastructureFraction(header, statedb)
+		gasPriceMinimum, _ := gpm.GetGasPriceMinimum(tx.GasCurrency(), header, statedb)
+		receipt, _, err := ApplyTransaction(p.config, p.bc, nil, gp, header, statedb, tx, usedGas, cfg, gasPriceMinimum, infraFraction)
```

Comment:
I think this is a great idea. I spent some time working on this though and ran into an issue with accessing the stateDB held by the evm object (since it is type `vm.StateDB` instead of the `state.StateDB` type we need). Do you feel okay punting on this?

Reply:
Ya, I was thinking about that.  So I am wondering if we could extend the vm.evm object to return a statedb and header, as it is associated with instances of those during it's creation.  (Note that I just requested that you delete a getStateDB method from the vm.evm object, so I may be retracting that.  Sorry ... ).

Reply:
done